### PR TITLE
Adding acceleration scaling factor

### PR DESCRIPTION
**Pull request by ghost**
Extending changes in #547 to add acceleration scaling factor.  

This adds support for setting a maximum acceleration scaling factor in the MotionPlanRequest msg, which can be used to scale the acceleration bounds in add_time_parametrization.

Includes necessary changes to
    move_group
    move_group_interface
    Spinbox for motion planning rviz plugin

Requires other pulls requests in moveit_msgs and moveit_core which are stated below.

### CODE REVIEW

**Comment by ghost** [comment]
I noticed I had a few naming issues in motion_planning_rviz_plugin_frame.ui that would generate warnings during the build process.  These were corrected in commit 5a18d6e5f7256d6bf9ea97969297f4ee3afbab8d

File: visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui [hunk 1]
```diff
@@ -628,7 +628,31 @@
...
+             </widget>
+            </item>
+           </layout>
+          </item>
+          <item>
+           <layout class="QHBoxLayout" name="horizontalLayout_20">
+            <item>
+             <widget class="QLabel" name="label_20">
+              <property name="text">
+               <string>Acceleration Scaling:</string>
```
Comment (davetcoleman) [inline visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui:650]: Does this long name widen the UI too much? Screenshot?

**Comment by davetcoleman** [comment]
lgtm +1

File: visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui (hunk 1 above)
Comment (davetcoleman) [inline visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui:650]: Oh, I just saw your [screenshot](https://cloud.githubusercontent.com/assets/9295956/12263573/b2b92128-b8f5-11e5-9b36-884760dec9cf.png).

If you change this to "Accel Scaling" does it reduce the required screen width? The MoveIt widget is already too wide IMHO so I'd prefer not to do something to make it even wider

Comment (ghost) [inline visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui:650]: I agree with you and thing the whole widget is a bit big.  As for the addition, I can do a side by side comparison but I am pretty sure that the widget width went unchanged.

Comment (ghost) [inline visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui:650]: Before and after.  No change in width but the widget did get one line taller (expected).

screenshot from 2016-01-14 11 29 47

After

**Comment by awesomebytes** [comment]
Just a tiny comment: This is awesome. Looking forward to merge it (and maybe release?).

**Comment by davetcoleman** [comment]
@hemes could you fix the merge conflicts? sorry about that.

**Comment by ghost** [comment]
@davetcoleman ok, I got the conflicts fixed.  In the process I had the same issue as last time where all of the commits that my fork was behind were added as commits to my fork.  So I deleted the commit and got it working the second time but in the process seem to have messed up the commit history for the pull request.  The final result is correct now.

BTW is there a good reference or best practice for keeping up to date with upstream repositories?  I set the original repo as the upstream remote but am unclear on the proper commands to update and get changes that have happened between the pull request and now (i.e., when to pull, merge, rebase, etc.).

**Comment by davetcoleman** [comment]
re:BTW - I maintain my indigo-devel to mirror the upstream version exactly, I never commit new stuff to that branch. Then I can just do `git pull upstream indigo-devel` and then merge that updated branch to whatever feature branches I have. Note that by `upstream` I mean `ros-planning`

**Pull request merged**